subsys/net/l2/ethernet/bridge.c

```diff
@@ -94,11 +94,18 @@ int eth_bridge_iface_add(struct net_if *br, struct net_if *iface)
 	int count = 0;
 	int ret;
 
+#if defined(CONFIG_NET_DSA) && !defined(CONFIG_NET_DSA_DEPRECATED)
```

Comment:
Should these checks be done at runtime? I mean it should be possible to have normal Ethernet ports used in a device that has DSA enabled.